CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/stream/ExtendedBufferNGTest.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package au.gov.asd.tac.constellation.utilities.stream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.reflect.Field;
+import static org.testng.Assert.*;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author serpens24
+ */
+public class ExtendedBufferNGTest {
+    
+    final private int size = 10;
+    
+    public ExtendedBufferNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test the default constructor - essentially just show it creates the
+     * 
+     */
+    @Test
+    public void testDefaultConstructor() throws Exception {
+        System.out.println("ExtendedBufferNGTest.testDefaultConstructor");
+
+        final ExtendedBuffer localBuffer = new ExtendedBuffer(size);
+        final Field bufferSize = ExtendedBuffer.class.getDeclaredField("bufferSize");
+        bufferSize.setAccessible(true);
+        assertEquals(size, (int)bufferSize.get(localBuffer));
+        assertEquals(localBuffer.getAvailableSize(), 0);
```

Comment:
I'm ok with the use of reflection here but the other option would be to add a getter for the bufferSize.

Reply:
yeah I think a protected getter might be a better option in this instance

Reply:
Yep thats an easy add.